app/mender.go

```diff
@@ -682,19 +685,39 @@ func (m *Mender) InventoryRefresh() error {
 		{Name: "mender_client_version", Value: conf.VersionString()},
 	}
 
-	if idata == nil {
-		idata = make(client.InventoryData, 0, len(reqAttr))
+	if iData == nil {
+		iData = make(client.InventoryData, 0, len(reqAttr))
 	}
-	_ = idata.ReplaceAttributes(reqAttr)
+	_ = iData.ReplaceAttributes(reqAttr)
 
-	if idata == nil {
-		log.Infof("No inventory data to submit")
+	if iData == nil {
+		log.Info("No inventory data to submit")
 		return nil
 	}
 
-	err = ic.Submit(m.api.Request(m.authToken, nextServerIterator(m), reauthorize(m)), m.Config.Servers[0].ServerURL, idata)
+	// sort slice to have a reproducible hash
+	sort.Sort(client.ByName(iData))
+
+	iDataBuf, err := json.Marshal(iData)
+	if err != nil {
+		return err
+	}
+	iDataSha := sha256.Sum256(iDataBuf)
+	lastInvSha, err := m.GetCurrentInventoryHash()
+	if err == nil && iDataSha == lastInvSha {
+		log.Info("Inventory data has not changed, skip submit")
```

Comment:
🤔 
Logging is always hard. I don't particularly like this message, but I'm also struggling with finding something "better":
`The inventory data has not changed since the last time it was collected. Nothing to submit. Skipping server inventory update call`. 
Or something along these lines?

Reply:
You suggestion was a bit wordy, I cut the middle part:
`The inventory data has not changed since the last time it was collected. Skipping server inventory update call`

Reply:
Nice! A little back and forth usually does the trick with these